framework/src/packets/ip/v6/srh.rs

```diff
@@ -168,8 +176,31 @@ impl<E: Ipv6Packet> SegmentRouting<E> {
         self.header().segments_left
     }
 
-    #[inline]
-    pub fn set_segments_left(&mut self, segments_left: u8) {
+    /// Sets the segments left and the IPv6 `dst` to match.
+    ///
+    /// Given a segment list of 3 segments, `[a, b, c]`, if the
+    /// `segments_left` is set to 2, then the IPv6 `dst` is set
+    /// to the (n + 1)th segment in the list, which is `c`.
+    ///
+    /// # Remarks
+    ///
+    /// `segments` should be set before setting `segments_left`
+    /// because the new `dst` value is taken from the segments
+    /// list.
+    #[inline]
+    pub fn set_segments_left(&mut self, segments_left: u8) -> Result<()> {
+        if let Some(&segment) = self.segments().get(segments_left as usize) {
+            self.header_mut().segments_left = segments_left;
+            self.envelope_mut().set_dst(IpAddr::V6(segment))
+        } else {
+            Err(SegmentsLeftOutOfBoundError(segments_left, self.segments().len()).into())
+        }
+    }
+
+    // hack, internal setter intended for proptest packet generator
+    // to skip the consistency check. do not use otherwise.
+    #[cfg(feature = "test")]
+    pub(crate) fn internal_set_segments_left(&mut self, segments_left: u8) {
```

Comment:
I'd think that _set_segments_left is more canonical personally (as an internal fn) than using the word internal. Also, should we doc hidden this (will hit hide the fn completely right?).

Reply:
ok, added doc hidden and changed to leading double underscore `__set_segments_left`. I've seen that used somewhere as well so seems legit.